docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -2008,6 +2008,157 @@ against those packages and its datatypes are not automatically converted.
+should not be part of your model and should not be uploaded to the ledger.
+
+Daml Script packages are not guaranteed to be upgradeable across SDK versions.
+If you depend on a Daml Script datatype in a serializable
+position (e.g. the field of a template), your package may rely on
+a Daml Script package in a way that can neither be removed nor upgraded to the
+next SDK version. Your package and any of its SCU upgrades would
+be stuck on that SDK version.
+
+For example, suppose you have a ``main`` package that depends on
+``daml-script-lts`` from SDK 2.10.0.
+
+.. code:: yaml
+
+  version: 2.10.0
+  name: main
+  ...
+  dependencies:
+  - daml-prim
+  - daml-stdlib
+  - daml-script-lts
+  build-options:
+  - --target=1.17
+
+.. code:: daml
+
+  module Main where
+
+  import qualified Daml.Script
+
+  data MyData = MyData
+    { field1 : Daml.Script.PartyIdHint
+    }
+
+Because ``MyData`` is a serializable datatype, any changes to it must be valid
+upgrade changes (e.g. adding a field) for the ``main`` package itself to be upgraded. If SDK 2.10.X introduces a change to
+``daml-script-lts`` that is not a valid upgrade of the ``daml-script-lts`` in
+SDK 2.10.0, then ``field1`` is not upgradeable to the next version of the
+SDK, nor can the field be dropped because the field is used in SCU checks.
+
+**Note**: While it is unlikely that an SDK update breaks ``daml-script-lts``
+for upgrades, we still strongly recommend against it. In Daml 3.x, Canton may
+disallow uploading ``daml-script-lts`` entirely.
+
+At that point, all future development on ``main``, including future upgrades,
+would be locked to SDK 2.10.0. To bump the SDK version, ``main`` would have to
+be migrated via a manual upgrade tool with downtime -- existing downtime upgrade
+techniques are listed :ref:`here <upgrades-index>`.
+
+For this reason, we strongly recommend against using Daml Script as an upgrade
+dependency for any package going on the ledger. Whenever building the ``main``
+package above, you can expect a warning:
+
+.. code:: bash
+
+  > daml build
+  ...
+  warning while type checking data type Main.MyData:
+    This package depends on a datatype <package ID>:Daml.Script.Stable:PartyIdHint from <package ID> (daml-script-lts-stable, 0.0.0) with LF version 1.17.
+    
+    We do not recommend using datatypes from Daml Script in >= LF1.17 packages, because those datatypes are not upgradeable.
+    Upgrade this warning to an error -Werror=upgrade-serialized-daml-script
+    Remove this warning with -Wno-upgrade-serialized-daml-script
+
+**Note:** For added safety, you may upgrade these warnings to errors with
+``-Werror=upgrade-serialized-daml-script``. We recommend against removing these
+warnings with ``-Wno-upgrade-serialized-daml-script``.
+
+If instead ``main`` depends on a datatype in a non-serializable position such
+as the type signature of a function, ``main`` can still be upgraded without
+breaking any SCU restrictions. For example:
```

Comment:
I'm wondering if we shouldn't drop that part. It may give a false sense of security to the users. There's still the problem that if you upload two different versions of daml-script-lts to the ledger, they may not be compatible. Then again, if we ever break that, we should ear our own dogfood and call it daml-script-lts-v2.

Reply:
I've been playing with rewriting that section, and couldn't find anything better than a the bolded Note approach that we already have. I don't want to unmention it - I'd rather customers be aware of a workaround, along with its caveats, than that they discover it on their own and think it's a solution with no drawbacks.